cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/sink/DBSink.java

```diff
@@ -168,25 +170,53 @@ private void setResultSetMetadata() throws Exception {
    *
    * @throws Exception if the driver is not available
    */
-  private void ensureJDBCDriverIsAvailable() throws Exception {
+  private void ensureJDBCDriverIsAvailable(Class<? extends Driver> jdbcDriverClass) throws Exception {
     try {
       DriverManager.getDriver(dbSinkConfig.connectionString);
     } catch (SQLException e) {
       // Driver not found. We will try to register it with the DriverManager.
       LOG.debug("Plugin Type: {} and Plugin Name: {}; Driver Class: {} not found. Registering JDBC driver via shim {} ",
-                dbSinkConfig.jdbcPluginType, dbSinkConfig.jdbcPluginName, driverClass.getName(),
+                dbSinkConfig.jdbcPluginType, dbSinkConfig.jdbcPluginName, jdbcDriverClass.getName(),
                 JDBCDriverShim.class.getName());
-      driverShim = new JDBCDriverShim(driverClass.newInstance());
-      DBUtils.deregisterAllDrivers(driverClass);
+      if (driverShim == null) {
+        driverShim = new JDBCDriverShim(jdbcDriverClass.newInstance());
+      }
+      DBUtils.deregisterAllDrivers(jdbcDriverClass);
       DriverManager.registerDriver(driverShim);
     }
   }
 
+  private boolean tableExists(Class<? extends Driver> jdbcDriverClass) {
+    try {
+      ensureJDBCDriverIsAvailable(jdbcDriverClass);
+    } catch (Exception e) {
+      throw new IllegalArgumentException("JDBCDriver is not available", e);
+    }
+    try (Connection connection = createConnection();
+         ResultSet rs = connection.getMetaData().getTables(null, null, dbSinkConfig.tableName, null)
+    ) {
+      return rs.next();
+    } catch (java.sql.SQLException e) {
+      throw new IllegalArgumentException("SQL Exception thrown when trying to connect to driver", e);
+    } finally {
+      DBUtils.cleanup(jdbcDriverClass);
```

Comment:
Since the `jdbcDrvierClass` is passed to this class, the cleanup should be done from the caller side.